packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/state_var.rs

```diff
@@ -535,17 +559,66 @@ impl<T: Default + Clone> StateVarReadOnlyView<T> {
     }
 }
 
+#[enum_dispatch]
+pub trait QueryUpdateRequests {
+    /// Reset variable that tracks whether or not an update has been requested
+    fn reset_queued_updates(&mut self);
+
+    fn return_indices_with_queued_updates(&self) -> Vec<usize>;
+}
+
+impl<T: Default + Clone> QueryUpdateRequests for StateVarReadOnlyView<T> {
+    /// Reset 'update_has_been_queued` to false
+    fn reset_queued_updates(&mut self) {
+        self.update_has_been_queued = false;
+    }
+
+    fn return_indices_with_queued_updates(&self) -> Vec<usize> {
+        if self.update_has_been_queued {
+            vec![0]
+        } else {
+            vec![]
+        }
+    }
+}
+
+impl<T> QueryUpdateRequests for Vec<T>
+where
+    T: QueryUpdateRequests,
+{
+    fn reset_queued_updates(&mut self) {
+        for val in self.iter_mut() {
+            val.reset_queued_updates();
+        }
+    }
+
+    fn return_indices_with_queued_updates(&self) -> Vec<usize> {
+        // Note: this algorithm does not correctly handle Vec<Vec<T: QueryUpdateRequests>>
+        self.iter()
+            .enumerate()
+            .filter_map(|(idx, val)| {
+                if val.return_indices_with_queued_updates().is_empty() {
```

Comment:
Seems like you should have returned an option to begin with...

Reply:
Not sure how `Option<Vec<T>>` is better than `Vec<T>`. In the former case, both `None` and `Some(vec![])` would represent the state of having nothing.